subsys/dfu/img_util/flash_img.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __FLASH_IMG_H__
+#define __FLASH_IMG_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct flash_img_context {
+	u8_t buf[CONFIG_IMG_BLOCK_BUF_SIZE];
+	struct device *dev;
+	u32_t bytes_written;
+	u16_t buf_bytes;
+};
+
+/**
+ * @brief Initialize context needed for writing the image to the flash.
+ *
+ * @param ctx context to be initialized
+ * @param dev flash driver to used while writing the image
+ */
+void flash_img_init(struct flash_img_context *ctx, struct device *dev);
+
+/**
+ * @brief Read number of bytes of the image written to the flash.
+ *
+ * @param ctx context
+ *
+ * @return Number of bytes written to the image flash.
+ */
+size_t flash_img_bytes_written(struct flash_img_context *ctx);
+
+/**
+ * @brief  Process input buffers to be written to the image slot 1. flash
+ * memory in single blocks. Will store remainder between calls.
+ *
+ * A final call to this function with flush set to true
+ * will write out the remaining block buffer to flash. As One block is always
```

Comment:
Thanks for fixing this! Note that "One" should be lower-cased.

(You might want to say something like "Since flash is written to in blocks, the contents of flash from the last byte written up to the next multiple of CONFIG_IMG_BLOCK_BUF_SIZE is padded with 0xff".)